core/protocols/aws-json-protocol/src/test/java/software/amazon/awssdk/protocols/json/FaultStatusCodeMappingTest.java

```diff
@@ -66,7 +90,17 @@ public static List<TestCase> unmarshal_faultValue_testCases() {
         );
     }
 
-    private static AwsJsonProtocolErrorUnmarshaller makeUnmarshaller(List<ExceptionMetadata> exceptionMetadata) {
+    public static List<QueryErrorTestCase> x_amzn_query_error_testCases() {
+        return Arrays.asList(
+            new QueryErrorTestCase("actualErrorCode;Sender", "actualErrorCode"),
+            new QueryErrorTestCase("actualError CodeSender", "ServiceException"),
+            new QueryErrorTestCase("actualError", "ServiceException"),
```

Comment:
Just a query , why delimeter ';' so much important here , why cant we take the error message whatever we find when delimeter doesnot exist like `actualError` in this case.
Anyways I donot see the text following the delimter is used by SDK in this case ;Sender the Sender is used not used anywhere.

Reply:
Yes, I'm just following the rule made by the SEP. the semi colon is a hard requirement. So if the header only contains the errorCode (without the delimiter or error type), sdk should fall back parsing the errorCode from its original source (either from content or something else). The `;Sender` is indeed not used in Java sdk but in some other sdk, it does exposed to the customer.